### DISCUSSION TITLE
[Modal] long - bodyMargin adds unecessary margin-right

### DISCUSSION THREAD

**Discussion by samydoesit** [fomantic/Fomantic-UI#749] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
# Help Wanted
Modal Dimmer adds unecessary `margin-right` on the body element.
## Problem

When the Dimmer appears the bodyMargin function inside `semantic.js:11692` adds in our case an unecessary margin-right on the body element. Is there any way to control this behaviour without commenting the line out?

**Comment by lubber-de** [comment, marked as answer]
This is a wanted behavior to avoid the hard visual movement of the body and modal content when a scrollbar appears invented by #468 .

The margin-right value should be set/calculated dynamically, so in case there is no scrollbar set when the page dimmer opens for a modal, the margin-right should be 0, thus not having any visual negative side effect.

Could you please provide a jsfiddle for your case where this behavior is disturbing?

**Comment by samydoesit** [comment]
Thanks for the quick reply.

Ok now i understand this, thanks to #468.

We got a bad css rule that was inheriting the overflow to the body and broke it, now everything is working fine thanks!
